pkg/provider/aws/parameterstore/parameterstore_test.go

```diff
@@ -37,216 +37,158 @@ func TestConstructor(t *testing.T) {
-				Parameter: &ssm.Parameter{
-					Value: aws.String("RRRRR"),
-				},
-			},
-			apiErr:         nil,
-			expectError:    "",
-			expectedSecret: "RRRRR",
-		},
-		{
-			// good case: extract property
-			apiInput: &ssm.GetParameterInput{
-				Name:           aws.String("/baz"),
-				WithDecryption: aws.Bool(true),
-			},
-			rr: esv1alpha1.ExternalSecretDataRemoteRef{
-				Key:      "/baz",
-				Property: "/shmoo",
-			},
-			apiOutput: &ssm.GetParameterOutput{
-				Parameter: &ssm.Parameter{
-					Value: aws.String(`{"/shmoo": "bang"}`),
-				},
-			},
-			apiErr:         nil,
-			expectError:    "",
-			expectedSecret: "bang",
-		},
-		{
-			// bad case: missing property
-			apiInput: &ssm.GetParameterInput{
-				Name:           aws.String("/baz"),
-				WithDecryption: aws.Bool(true),
-			},
-			rr: esv1alpha1.ExternalSecretDataRemoteRef{
-				Key:      "/baz",
-				Property: "INVALPROP",
-			},
-			apiOutput: &ssm.GetParameterOutput{
-				Parameter: &ssm.Parameter{
-					Value: aws.String(`{"/shmoo": "bang"}`),
-				},
-			},
-			apiErr:         nil,
-			expectError:    "key INVALPROP does not exist in secret",
-			expectedSecret: "",
-		},
-		{
-			// bad case: extract property failure due to invalid json
-			apiInput: &ssm.GetParameterInput{
-				Name:           aws.String("/baz"),
-				WithDecryption: aws.Bool(true),
-			},
-			rr: esv1alpha1.ExternalSecretDataRemoteRef{
-				Key:      "/baz",
-				Property: "INVALPROP",
-			},
-			apiOutput: &ssm.GetParameterOutput{
-				Parameter: &ssm.Parameter{
-					Value: aws.String(`------`),
-				},
-			},
-			apiErr:         nil,
-			expectError:    "key INVALPROP does not exist in secret",
-			expectedSecret: "",
-		},
-		{
-			// case: parameter.Value may be nil but binary is set
-			apiInput: &ssm.GetParameterInput{
-				Name:           aws.String("/baz"),
-				WithDecryption: aws.Bool(true),
-			},
-			rr: esv1alpha1.ExternalSecretDataRemoteRef{
-				Key: "/baz",
-			},
-			apiOutput: &ssm.GetParameterOutput{
-				Parameter: &ssm.Parameter{
-					Value: nil,
-				},
-			},
-			apiErr:         nil,
-			expectError:    "parameter value is nil for key",
-			expectedSecret: "",
-		},
-		{
-			// should return err
-			apiInput: &ssm.GetParameterInput{
-				Name:           aws.String("/foo/bar"),
-				WithDecryption: aws.Bool(true),
-			},
-			rr: esv1alpha1.ExternalSecretDataRemoteRef{
-				Key: "/foo/bar",
-			},
-			apiOutput:   &ssm.GetParameterOutput{},
-			apiErr:      fmt.Errorf("oh no"),
-			expectError: "oh no",
-		},
-	} {
-		f.WithValue(row.apiInput, row.apiOutput, row.apiErr)
-		out, err := p.GetSecret(context.Background(), row.rr)
-		if !ErrorContains(err, row.expectError) {
-			t.Errorf("[%d] unexpected error: %s, expected: '%s'", i, err.Error(), row.expectError)
+
+	// good case: extract property
+	setExtractProperty := func(pstc *parameterstoreTestCase) {
+		pstc.apiOutput.Parameter.Value = aws.String(`{"/shmoo": "bang"}`)
+		pstc.expectedSecret = "bang"
```

Comment:
What's missing here is:
```go
pstc.remoteRef.Property = "/shmoo"
```

once this is in and `expectedSecret` is being used below the coverage & functionality should be the same as before!